src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -540,9 +561,117 @@ private List<GeneratorTemplate> getModElementGeneratorTemplatesList(ModElement e
+			GeneratableElement generatableElement) {
+
+		if (generatableElement == null) {
+			generatableElement = element.getGeneratableElement();
+			if (generatableElement == null) { // we can't construct list data because we have nothing to process
+				LOG.warn("Failed to load mod generatable element: " + element.getName()
+						+ ". This means no list templates will be generated");
+				return new ArrayList<>();
+			}
+		}
+
+		Map<?, ?> map = generatorConfiguration.getDefinitionsProvider().getModElementDefinition(element.getType());
+		if (map == null) {
+			LOG.info("Failed to load element definition for mod element type " + element.getType().getRegistryName());
+			return null;
+		}
+
+		Set<GeneratorTemplatesList> fileLists = new HashSet<>();
+
+		List<?> templateLists = (List<?>) map.get("list_templates");
+		if (templateLists != null) {
+			int templateID = 0;
+			for (Object list : templateLists) {
+				Map<GeneratorTemplate, List<Boolean>> files = new LinkedHashMap<>();
+				String groupName = (String) ((Map<?, ?>) list).get("name");
+				Object listData = TemplateExpressionParser.processFTLExpression(this,
+						(String) ((Map<?, ?>) list).get("listData"), generatableElement);
+				List<?> templates = (List<?>) ((Map<?, ?>) list).get("forEach");
+				// we check type of list data collection and convert it to a list if needed
+				List<?> elements;
+				if (listData instanceof Map<?, ?> listMap)
+					elements = new ArrayList<>(listMap.entrySet());
+				else if (listData instanceof Collection<?> collection)
+					elements = new ArrayList<>(collection);
+				else if (listData instanceof Iterable<?> iterable) // fallback for the worst case
+					elements = StreamSupport.stream(iterable.spliterator(), false).collect(Collectors.toList());
+				else
+					elements = new ArrayList<>();
+				if (templates != null) {
+					for (Object template : templates) {
+						String rawname = (String) ((Map<?, ?>) template).get("name");
+
+						TemplateExpressionParser.Operator operator = TemplateExpressionParser.Operator.AND;
+						Object conditionRaw = ((Map<?, ?>) template).get("condition");
+						if (conditionRaw == null) {
+							conditionRaw = ((Map<?, ?>) template).get("condition_any");
+							operator = TemplateExpressionParser.Operator.OR;
+						}
+
+						// we store file generation conditions for current mod element
+						List<Boolean> conditionChecks = new ArrayList<>();
+						for (int i = 0; i < elements.size(); i++) {
+							conditionChecks.add(i, !TemplateExpressionParser.shouldSkipTemplateBasedOnCondition(this, conditionRaw, elements.get(i), operator));
+						}
+
+						if (!conditionChecks.contains(true) && performFSTasks)
+							continue;
+
+						// we check for potential excludes to be deleted
+						// this is only called if condition above is passed
+						String exclude = (String) ((Map<?, ?>) template).get("exclude");
+						boolean doExclude = "true".equals(((Map<?, ?>) template).get("excludeIfAllPresent")) ?

```

Comment:
What is this `excludeIfAllPresent` parameter? I think exclude system should not be possible with list templates at all as it brings more problems than solutions.

In fact I don't think we use it in generator even for normal templates anywhere and could probably be entirely removed.

Reply:
> What is this `excludeIfAllPresent` parameter?

All is said in the pull request description.
> In fact I don't think we use it in generator even for normal templates anywhere

If this is on your TODO, I don't see a big problem in removing this parameter.